shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1681,11 +1687,36 @@ - (CGRect)firstRectForRange:(UITextRange*)range {
         !isLastSelectionRect && _selectionRects[i + 1].position > first;
     if (startsOnOrBeforeStartOfRange &&
         (endOfTextIsAfterStartOfRange || nextSelectionRectIsAfterStartOfRange)) {
-      return _selectionRects[i].rect;
+      // TODO(hellohaunlin): Remove iOS 17 check. The logic should also work for older versions.
+      if (@available(iOS 17, *)) {
+        startSelectionRect = _selectionRects[i].rect;
+      } else {
+        return _selectionRects[i].rect;
+      }
+    }
+    // TODO(hellohaunlin): Remove iOS 17 check. The logic should also work for older versions.
+    if (@available(iOS 17, *)) {
```

Comment:
You can remove this one now; CGRectIsNull will always be true for <17 because of the check you already have above.